pkg/standalone-utils/contracts/ProtocolFeeSplitter.sol

```diff
@@ -0,0 +1,162 @@
+
+    // poolId => PoolSettings
+    mapping(bytes32 => RevenueShareSettings) public poolSettings;
+
+    constructor(IProtocolFeesWithdrawer _protocolFeesWithdrawer, address _treasury)
+        // The ProtocolFeeSplitter is a singleton, so it simply uses its own address to disambiguate action
+        // identifiers.
+        Authentication(bytes32(uint256(address(this))))
+    {
+        protocolFeesWithdrawer = _protocolFeesWithdrawer;
+        treasury = _treasury;
+        vault = _protocolFeesWithdrawer.getProtocolFeesCollector().vault();
+    }
+
+    function setRevenueSharingFeePercentage(bytes32 poolId, uint256 newSwapFeePercentage)
+        external
+        override
+        authenticate
+    {
+        _require(newSwapFeePercentage <= _MAX_REVENUE_SHARING_FEE_PERCENTAGE, Errors.SPLITTER_FEE_PERCENTAGE_TOO_HIGH);
+        poolSettings[poolId].revenueSharePercentageOverride = uint96(newSwapFeePercentage);
+        emit PoolRevenueShareChanged(poolId, newSwapFeePercentage);
+    }
+
+    function setDefaultRevenueSharingFeePercentage(uint256 feePercentage) external override authenticate {
+        _require(feePercentage <= _MAX_REVENUE_SHARING_FEE_PERCENTAGE, Errors.SPLITTER_FEE_PERCENTAGE_TOO_HIGH);
+        defaultRevenueSharingFeePercentage = feePercentage;
+        emit DefaultRevenueSharingFeePercentageChanged(feePercentage);
+    }
+
+    function setTreasury(address newTreasury) external override authenticate {
+        treasury = newTreasury;
+        emit TreasuryChanged(newTreasury);
+    }
+
+    function setPoolBeneficiary(bytes32 poolId, address newBeneficiary) external override {
+        (address pool, ) = vault.getPool(poolId);
+        _require(msg.sender == Pool(pool).getOwner(), Errors.SENDER_NOT_ALLOWED);
+        poolSettings[poolId].beneficiary = newBeneficiary;
+        emit PoolBeneficiaryChanged(poolId, newBeneficiary);
+    }
+
+    function collectFees(bytes32 poolId) external override {
+        (address pool, ) = vault.getPool(poolId);
+        IERC20 bpt = IERC20(pool);
+
+        IProtocolFeesWithdrawer protocolFeesWithdrawerMemory = protocolFeesWithdrawer;
+        uint256 feeCollectorBptBalance = bpt.balanceOf(
+            address(protocolFeesWithdrawerMemory.getProtocolFeesCollector())
+        );
+        if (feeCollectorBptBalance == 0) {
+            _revert(Errors.NO_BPT_FEES_COLLECTED);
+        }
+
+        // withdrawCollectedFees requires array of tokens, and we are only transfering BPT token
+        IERC20[] memory tokens = new IERC20[](1);
+        tokens[0] = bpt;
+
+        address beneficiary = poolSettings[poolId].beneficiary;
+
+        if (beneficiary == address(0)) {
+            uint256[] memory amounts = new uint256[](1);
+            amounts[0] = feeCollectorBptBalance;
+            protocolFeesWithdrawerMemory.withdrawCollectedFees(tokens, amounts, treasury);
+            emit FeesCollected(poolId, beneficiary, 0, treasury, feeCollectorBptBalance);
+        } else {
```

Comment:
This duplication seems unnecessary - you could instead have the following function

```
fn getPoolOwnerFeePercentage() {
  if beneficiary == 0
    return 0

  if feeOverride != 0
    return fee override
  else
   return default
} 
```

then pass that to `_computeAmounts`, and have the suggested `withdraw` helper simply return immediately if `amount` is 0. You'll end up with fewer branches and less special-casing - plus, the above function is a good `public` getter anyway, since otherwise it's relatively hard to compute from the outside what those might be.

I'd consider even going one step further and providing a second getter that returns the amounts that could be collected to the owner and to the treasury if `collect` was called now.